lua/orgmode/state/state.lua

```diff
@@ -0,0 +1,167 @@
+local utils = require('orgmode.utils')
+local Promise = require('orgmode.utils.promise')
+
+local State = { data = {}, _ctx = { loaded = false, saved = false, curr_loader = nil, savers = 0 } }
```

Comment:
Add a type to the class so it's properly read by lua ls. You can also rename everything from `State` to `OrgState`, but not necessary. I just want to make type prefixed to not potentially clash with other plugins.
```suggestion
---@class OrgState
local State = { data = {}, _ctx = { loaded = false, saved = false, curr_loader = nil, savers = 0 } }
```

Reply:
Done. I also renamed the `State` object to `OrgState`.